packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -134,8 +135,18 @@ class MessagePayload {
     }
 
     const enforce_nonce = Boolean(this.options.enforceNonce);
-    if (enforce_nonce && nonce === undefined) {
-      throw new DiscordjsError(ErrorCodes.MessageNonceRequired);
+
+    /**
+     * If `nonce` isn't provided, generate one & set `enforceNonce`
+     * Unless `enforceNonce` is explicitly set to `false`(not just falsy)
+     */
+    if (nonce === undefined) {
+      if (this.options.enforceNonce !== false && this.client.options.enforceNonce) {
+        this.options.nonce = DiscordSnowflake.generate().toString();
+        this.options.enforceNonce = true;
+      } else if (enforce_nonce) {
+        throw new DiscordjsError(ErrorCodes.MessageNonceRequired);
+      }
```

Comment:
IMO we should just get rid of this[^1] and automatically generate a nonce if none is passed when `enforceNonce` is set to `true`.
Probably results in better DX and simplifies the logic a bit 

[^1]: meaning we stop throwing the error and deprecate the error code

Reply:
While I don't disagree with you, I think that would end up changing the semver for this particular PR. What do you think?

Reply:
Lets not change the current behavior, although technically not breaking in this case.